src/app-layer-detect-proto.c

```diff
@@ -1395,13 +1395,13 @@ static int AppLayerProtoDetectPMPrepareMpm(AppLayerProtoDetectPMCtx *ctx)
     SCReturnInt(ret);
 }
 
-static void AppLayerProtoDetectPMFreeSignature(AppLayerProtoDetectPMSignature *sig)
+static void AppLayerProtoDetectPMFreeSignature(DetectEngineCtx *de_ctx, AppLayerProtoDetectPMSignature *sig)
```

Comment:
we have no detect engine in the proto detection, so no need to pass it around. Just pass NULL into DetectContentFree